src/runtime/PromiseObject.cpp

```diff
@@ -325,4 +326,69 @@ Value promiseAllResolveElementFunction(ExecutionState& state, Value thisValue, s
     }
     return Value();
 }
+
+Value promiseThenFinally(ExecutionState& state, Value thisValue, size_t argc, Value* argv, bool isNewExpression)
+{
+    // Let onFinally be F.[[OnFinally]].
+    Value onFinally = argv[1];
+    Value result;
+
+    // Assert: IsCallable(onFinally) is true.
+    // Let result be ? Call(onFinally, undefined).
+    ASSERT(onFinally.asPointerValue()->isCallable());
+    result = Object::call(state, onFinally, Value(), 0, &thisValue);
+
+    // Let C be F.[[Constructor]].
+    // Assert: IsConstructor(C) is true.
+    Value C = argv[0];
+    ASSERT(C.asPointerValue()->isCallable());
+
+    Value promise = promiseResolve(state, C.asObject(), result);
+
+    // Let valueThunk be equivalent to a function that returns value.
+    // Return ? Invoke(promise, "then", « valueThunk »).
+    Object* thisObject = thisValue.toObject(state);
+    auto strings = &state.context()->staticStrings();
+    Value then = thisObject->get(state, strings->then).value(state, thisObject);
+    Value argument[1] = { Value(valueThunkHelper(argv[3])) };
```

Comment:
That's definitely not a good way to go. The expression `valueThunkHelper(argv[3])` equals to `argv[3]`. Moreover the argv[1] should be passed to `valueThunk`(This resolves automatically by using `PromiseStepFunctionObject`).

Secondly, the standard says `Let valueThunk be equivalent to a function that returns value.`

This can be performed like this:
 - We should also create a new `ValueThunkFunctionObject` inherited from `NativeFunctionObject`, with a member `m_value`.
- When the function is called (~ 10 lines below), the function should also resolve the active function object (which should be `ValueThunkFunctionObject`, then return the stored `m_value`.)